include/irr/asset/CQuantNormalCache.h

```diff
@@ -363,6 +340,76 @@ class CQuantNormalCache
 	core::unordered_map<VectorUV, uint32_t, QuantNormalHash, QuantNormalEqualTo> normalCacheFor2_10_10_10Quant;
 	core::unordered_map<VectorUV, Vector8u, QuantNormalHash, QuantNormalEqualTo> normalCacheFor8_8_8Quant;
 	core::unordered_map<VectorUV, Vector16u, QuantNormalHash, QuantNormalEqualTo> normalCacheFor16_16_16Quant;
+
+private:
+	class CReadBufferWrap
+	{
+	public:
+		CReadBufferWrap(const SBufferRange<ICPUBuffer>& _buffer)
+			:buffer(_buffer)
+		{
+			buffPtr = static_cast<uint8_t*>(buffer.buffer.get()->getPointer());
+		}
+
+		bool load(char* p, size_t sz)
+		{
+			//TODO: safety
+			memcpy(p, buffPtr, sz);
+			buffPtr += sz;
+
+			return true;
+		}
+
+		template<typename V>
+		typename std::enable_if<phmap::type_traits_internal::IsTriviallyCopyable<V>::value, bool>::type
+			load(V* v)
+		{
+			//TODO: safety
+			memcpy(reinterpret_cast<uint8_t*>(v), buffPtr, sizeof(V));
+			buffPtr += sizeof(V);
+
+			return true;
+		}
+
+	private:
+		const SBufferRange<ICPUBuffer>& buffer;
+		uint8_t* buffPtr;
+
+	};
+
+	class CWriteBufferWrap
+	{
+	public:
+		CWriteBufferWrap(SBufferBinding<ICPUBuffer>& _buffer)
+			:buffer(_buffer)
+		{
+			bufferPtr = static_cast<uint8_t*>(buffer.buffer.get()->getPointer());
+		}
+
+		bool dump(const char* p, size_t sz)
+		{
+			memcpy(bufferPtr, p, sz);
+			bufferPtr += sz;
+
+			return true;
+		}
```

Comment:
do the `load` and `dump` actually get called without polymorphism?

Reply:
yup, check out parameters of `load` and `dump` functions
https://github.com/greg7mdp/parallel-hashmap/blob/master/parallel_hashmap/phmap_dump.h#L120

Reply:
ah, sweet!